examples/Acoustics/compute_spectrum_production.py

```diff
@@ -0,0 +1,280 @@
+        plt.scatter(observer_id_all, oaspl_all, label="dBA")
+        plt.ylabel("OASPL [dBA]", fontsize=10)
+
+    else:
+        plt.scatter(observer_id_all, oaspl_all, label="dB")
+        plt.ylabel("OASPL [dB]", fontsize=10)
+
+    plt.xlabel("Observer ID", fontsize=10)
+    plt.grid(which="major")
+    plt.grid(which="minor", ls="--", color="k", linewidth=0.5)
+    plt.legend()
+    plt.savefig("figures/OASPL.png", dpi=500)
+    plt.close()
+
+
+def print_input_choices(observer_id, window, weighting, acoustics_data):
+    """This function prints the input choices to the screen"""
+
+    if observer_id == -1:
+        print("Computing aeroacoustic output for all observers")
+    else:
+        print("Computing aeroacoustic output for observer ID " + str(observer_id))
+        if observer_id >= len(acoustics_data) - 2:
+            print("Observer ID not present in the aeroacoustics output")
+            raise RuntimeError
+
+    if window == "Hann":
+        print("Hann Window")
+    elif window is None:
+        print("No Window")
+    else:
+        print("Invalid windowing option selected")
+        raise RuntimeError
+
+    if weighting == "A":
+        print("A-weighted output")
+    elif weighting is None:
+        print("Non-weighted output")
+    else:
+        print("Invalid weighting option selected")
+        raise RuntimeError
+
+
+def acoustics_output(input_file, output, window, weighting, observer_id):
+    """This function drives the acoustics output"""
+
+    oaspl_all = []
+    observer_id_all = []
+    os.makedirs("figures", exist_ok=True)
+
+    acoustics_data = read_csv_file(input_file)
+    print_input_choices(observer_id, window, weighting, acoustics_data)
+
+    if observer_id == -1:
+        n_observer = len(acoustics_data) - 2
+        observer_list = list(range(0, n_observer))
+    else:
+        observer_list = list(range(observer_id, observer_id + 1))
+
+    if weighting == "A":
+        with open(output, "w", encoding="utf-8") as f:
+            f.write("observer_id, dBA \n")
+
+    elif weighting is None:
+        with open(output, "w", encoding="utf-8") as f:
+            f.write("observer_id, dB \n")
+
+    for i_observer in observer_list:
+        print(f"Observer ID = {i_observer}")
+        # Assuming constant time step size for now
+        time_step_size = acoustics_data["time"][1] - acoustics_data["time"][0]
+        # Apply windowing function (even if no windowing is specified, then rectangular window applied)
+        windowed_pressure = apply_windowing_function(acoustics_data, i_observer, window)
+        # Compute spectrum and OASPL values
+        frequency, spl, oaspl = compute_spectrum(
+            windowed_pressure, time_step_size, weighting
+        )
+        if np.isnan(oaspl) == 0:
+            observer_id_all = np.append(observer_id_all, i_observer)
+            oaspl_all = np.append(oaspl_all, oaspl)
+
+        # Output CSV file containing OASPL values
+        if np.isnan(oaspl) == 0:
```

Comment:
can be simplified I think

if np.isnan(oaspl) is True:
    oaspl = 0
with open(output, "a", encoding="utf-8") as f:
    f.write(f"{i_observer}, {oaspl:.5f} \n")

Reply:
on second thought, I think you'll want to set `oaspl = 0` above when appending observers at ln. 260
`oaspl_all = np.append(oaspl_all, oaspl)`

Reply:
We want to print all the OASPL values to a file. If it's a NaN we want to print zero.
For the plotting however, we want to exclude zero values.